### PR TITLE
[ATTRIB] Attrib command improvements

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Fix remaining attrib command test failures

_Add additional functionality to attrib command._

JIRA issue: [CORE-16284](https://jira.reactos.org/browse/CORE-16284)

## attrib.c updates to add remaining functionality

_Attempt to add remaining unimplemented items to attrib command._

ReactOS_attrib_good

This should allow @ThFabba's functions mentioned below to work correctly.

Attrib_tests_ThFabba

ReactOS_attrib_good1

In addition there is now a very close symmetry between the "PrintAttribute" and "ChangeAttribute" functions of the program.

### CODE REVIEW

File: base/applications/cmdutils/attrib/attrib.c [hunk 1]
```diff
@@ -405,14 +403,33 @@ int wmain(int argc, WCHAR *argv[])
...
+        else if (!ChangeAttribute(szPath, szFileName, bRecurse, bDirectories, dwMask, dwAttrib))
         {
             ConResPrintf(StdOut, STRING_FILE_NOT_FOUND, argv[i]);
         }
     }
 
+// Code below handles the special case of 'attrib +h /s' and similar
+
+    if (bRecurse && dwMask && (wcscmp(szPath, L"") == 0))
+        {
```
Comment (katahiromz) [inline base/applications/cmdutils/attrib/attrib.c]: Indentation.

Comment (Doug-Lyons) [reply base/applications/cmdutils/attrib/attrib.c]: Fixed. Thanks.

File: base/applications/cmdutils/attrib/attrib.c [hunk 2]
```diff
@@ -105,46 +111,57 @@ PrintAttribute(
...
         }
 
         do
         {
             if (!(findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY))
                 continue;
 
-            if (!wcscmp(findData.cFileName, L".") ||
-                !wcscmp(findData.cFileName, L".."))
+            if (!wcscmp(findData.cFileName, L".") || !wcscmp(findData.cFileName, L".."))
```
[the remark this answers is not in the document]
Comment (Doug-Lyons) [reply base/applications/cmdutils/attrib/attrib.c]: Reverted.

File: base/applications/cmdutils/attrib/attrib.c [hunk 3]
```diff
@@ -105,46 +111,57 @@ PrintAttribute(
...
                 continue;
 
             wcscpy(pszFileName, findData.cFileName);
             wcscat(pszFileName, L"\\");
-            PrintAttribute(szFullName, pszFile, bRecurse);
+            bFound = PrintAttribute(szFullName, pszFile, bRecurse,
```
[the remark this answers is not in the document]
Comment (Doug-Lyons) [reply base/applications/cmdutils/attrib/attrib.c]: Changed.

**Comment by Doug-Lyons** [comment]
@HBelusca, Also see the   [PR#2141](https://github.com/reactos/reactos/pull/2141) for regressions tests for the attrib command. With these changes, there should be no failing tests left.

**Comment by Doug-Lyons** [comment]
Hello, @HBelusca. I know that you are busy, but I just wanted to note that it has been over a year since you self-assigned this to yourself. Please consider finding time to review it when you get the chance. Thanks. Doug Lyons

**Comment by Doug-Lyons** [comment]
@Extravert-ir, Thank you for your help in getting this committed.

**Comment by Doug-Lyons** [comment]
Thanks, @HBelusca, for fixing my typographical error by changing the "!=" to "|=". This fixes the last two failed tests in the cmd:attrib regression test.

**Pull request merged**